include/operations/SymbolicOperation.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include "Definitions.hpp"
+#include "Expression.hpp"
+#include "StandardOperation.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <optional>
+#include <unordered_set>
+#include <variant>
+
+namespace qc {
+    // Overload pattern for std::visit
+    template<typename... Ts>
+    struct Overload: Ts... {
+        using Ts::operator()...;
+    };
+    template<class... Ts>
+    Overload(Ts...) -> Overload<Ts...>;
```

Comment:
Didn't know that programming pattern. Nice one 👍🏼 

Just for reference: if we were to adopt C++20 at some point, the second line can actually be dropped, since the compiler deduces these rules automatically.

Reply:
Yeah only found out about it while working on this PR. Without it, Variants are just a bit awkward to work with (which they still are).